presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3ConfigurationUpdater.java

```diff
@@ -124,6 +128,12 @@ public void updateConfiguration(Configuration config)
         if (sseKmsKeyId != null) {
             config.set(S3_SSE_KMS_KEY_ID, sseKmsKeyId);
         }
+        if (webIdentityEnabled) {
+            config.setBoolean(S3_WEB_IDENTITY_ENABLED, true);
```

Comment:
nit: We can always set the config irrespective of true/false and set the token file only if `webIdentityEnabled` is true.

```suggestion
        config.setBoolean(S3_WEB_IDENTITY_ENABLED, webIdentityEnabled);
        if (webIdentityEnabled) {
```